adafruit_progressbar/__init__.py

```diff
@@ -0,0 +1,420 @@
+        space.
+        :return int:
+        """
+
+        return self.widget_height - self._get_fill_border_size()
+
+    def _get_fill_border_size(self):
+        """Determines any visual space reserved for the widget
+        based on the defined border thickness, and whether a margin
+        should be placed between the border and the bar.
+        The value is calculated as (2 x border_thickness) minus
+        (2 x margin_size). The value for margin_size is either 0 (zero)
+        or 1 (one) depending on the value of margin_size when the
+        widget was created.
+        :return int:
+        """
+
+        return (2 * self.border_thickness) + (2 * self.margin_size)
+
+    @property
+    def margin_size(self):
+        """Returns the size of the margin on a single side of the display
+        :return int:
+        """
+        return self._margin_size
+
+    @margin_size.setter
+    def margin_size(self, value):
+        """Sets the new size of the margin to be used between the border
+        (if displayed) and the value bar.
+
+        :param value: The new size of the margin between the border
+        and value bar on all sides of the widget.
+        :type value: int
+        """
+
+        assert isinstance(value, int), "The margin size must be an integer"
+
+        margin_spacing = (2 * value) + (2 * self._border_thickness)
+
+        assert margin_spacing < self.widget_width, (
+            "The size of the borders and margins combined can total the same or more"
+            "than the widget's width."
+        )
+
+        assert margin_spacing < self.widget_height, (
+            "The size of the borders and margins combined can total the same or more"
+            "than the widget's height."
+        )
+
+        self._margin_size = value
+        self._set_progress(self._progress)  # For a render pass
+
+    def get_value_ratio(self, value):
+        """Gets the ratio (percentage) of a given value within the
+        range of self.minimum and self.maximum.
+
+        :param value: The value for which the ration should be calculated
+        :type value: int/float
+
+        :return: The ratio of value:range
+        :rtype: float
+        """
+
+        if self.maximum == self.minimum:
+            return 0.0
+
+        return (float(value) - self.minimum) / (self.maximum - self.minimum)
+
+    def render(self, _old_value, _new_value, _progress_value) -> None:
+        """The method called when the display needs to be updated. This method
+        can be overridden in child classes to handle the graphics appropriately.
+
+        :param _old_value: The value from which we're updating
+        :type _old_value: float
+        :param _new_value: The value to which we're updating
+        :type _new_value: float
+        :param _progress_value: The value of the progress, or ratio between the new value and the
+            maximum value
+        :type _progress_value: float
+        :rtype None:
+        """
```

Comment:
I was tinkering with this a bit and I figured out that we can add this import:
```
from adafruit_progressbar.progressbar import ProgressBar
```
at the end of __init__.py below here.

If we do that it will make it so the original style imports in user code still work as well:
```
from adafruit_progressbar import ProgressBar
```

Reply:
I don't think this is desired as well, since optimally, users would choose between the `HorizontalProgressBar` and `VerticalProgressBar`. The `ProgressBar` class is there for backward compatibility.

I've thought about making `ProgressBar` the sole class, and then choose its visual style based on a property.That starts to really overload the class, and it seemed that it would make extending to other progress styles later more complex than it would be worth.

Reply:
Added to the CircuitPython weekly meeting's "In the weeds" discussion for the week of Feb 22, 2021